File: eva-core/src/main/java/io/nerv/cache/config/ValidatorConfiguration.java
package io.nerv.cache.config;

import org.hibernate.validator.HibernateValidator;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import javax.validation.Validation;
import javax.validation.Validator;
import javax.validation.ValidatorFactory;

/**
 * hibernate validator参数校验配置
 */
@Configuration
public class ValidatorConfiguration {
    @Bean
    public Validator validator(){
        ValidatorFactory validatorFactory = Validation.byProvider( HibernateValidator.class )
                .configure()
                .failFast(true)
                .buildValidatorFactory();

        return validatorFactory.getValidator();
    }
}

File: eva-core/src/main/kotlin/io/nerv/config/ValidatorConfiguration.kt
package io.nerv.config

import org.hibernate.validator.HibernateValidator
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import javax.validation.Validation
import javax.validation.Validator

/**
 * hibernate validator参数校验配置
 */
@Configuration
open class ValidatorConfiguration {
    @Bean
    open fun validator(): Validator {
        val validatorFactory = Validation.byProvider(HibernateValidator::class.java)
                .configure()
                .failFast(true)
                .buildValidatorFactory()
        return validatorFactory.validator
    }
}